inspire/modules/migrator/tasks.py

```diff
@@ -22,10 +22,66 @@
 
 """Manage migration from INSPIRE legacy instance."""
 
+from __future__ import absolute_import
+
+from invenio_ext.sqlalchemy import db
+from invenio_records.api import Record as record_api
+from invenio_records.models import Record
+
+from inspire.dojson.hep import hep
+from inspire.dojson.institutions import institutions
+from inspire.dojson.journals import journals
+from inspire.dojson.experiments import experiments
+from inspire.dojson.hepnames import hepnames
+from inspire.dojson.jobs import jobs
+from inspire.dojson.conferences import conferences
+from inspire.dojson.processors import _collection_in_record
+
+from dojson.contrib.marc21.utils import create_record as marc_create_record
+
+from lxml import etree
+
 from invenio_celery import celery
 
 
 @celery.task
-def migrate(url, **kwargs):
+def migrate(source, **kwargs):
     """Main migration function."""
-    pass
+    from dojson.contrib.marc21.utils import split_stream
+    with db.session.begin_nested():
+        for record in split_stream(open(source)):
```

Comment:
There is no Python native way of chunking but you can borrow something from: http://stackoverflow.com/questions/434287/what-is-the-most-pythonic-way-to-iterate-over-a-list-in-chunks

and then have:
```python
for chunk in chunks(split_stream(open(source)), 1000):
    with.db.session.begin_nested():
        for record in chunk:
            ...
    db.session.commit()
    bulk_enhance_records(...)
    bulk_elastic_index_records(...)
```